# Question No.3 
Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.


In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats

In [2]:
buyersratio=pd.read_csv("/content/BuyerRatio.csv")

In [3]:
buyersratio.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

#### **As we have 2 variables we can perform Chi-Square**

## **Chi-Square Test**

In [4]:
## H0 = The male-female buyer rations are similar across regions
## H1 = The male-female buyer rations are not similar across regions

In [5]:
from scipy import stats

In [6]:
buyersratio["East"]

0     50
1    435
Name: East, dtype: int64

In [7]:
buyersratio["West"]

0     142
1    1523
Name: West, dtype: int64

In [8]:
buyersratio["North"]

0     131
1    1356
Name: North, dtype: int64

In [9]:
buyersratio["South"]

0     70
1    750
Name: South, dtype: int64

In [10]:
buyersratio1=[buyersratio["East"], buyersratio["West"], buyersratio["North"], buyersratio["South"]]

In [11]:
stats.chi2_contingency(buyersratio1)

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

### **P value of Chi-Square test == 0.66**
###  0.66 > 0.05
## **H0 = The male-female buyer rations are similar across regions is Accepted**


# Question No. 4
     TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
from scipy.stats import chi2_contingency

In [13]:
orderform=pd.read_csv("/content/Costomer+OrderForm.csv")

In [14]:
orderform

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

###  **we have more than 2 variables so we can perform Chi-Square**

## Chi-Square Test

In [15]:
# H0 = The defective % does not varies by centre
# H1 = The defective % does varies by centre

In [16]:
orderform.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [17]:
orderform.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [18]:
orderform.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [19]:
orderform.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [20]:
# observed frequency = obsf

In [21]:
obsf=np.array([[271,267,269,280],[29,33,31,20]])
obsf                       # Make a contingency table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [22]:
chi2_contingency(obsf)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [23]:
chi2_contingency(obsf)[1]

0.2771020991233135

In [24]:
p_value=chi2_contingency(obsf)[1] 

In [25]:
p_value

0.2771020991233135

In [26]:
p_value > 0.05

True

### **P value of Chi-Square test = 0.277**

### **0.277 > 0.05**

## **H0 = The defective % does not varies by centre**  **is Accepted**